src/components/apps/launch/params/DiskResourceSelector.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Form field for displaying FileInput parameters.
+ *
+ * @author psarando
+ */
+import React from "react";
+
+import { useQuery } from "react-query";
+
+import { useTranslation } from "i18n";
+
+import InputSelector from "../InputSelector";
+
+import { ERROR_CODES, getErrorCode } from "components/utils/error/errorCode";
+
+import {
+    getResourceDetails,
+    DATA_DETAILS_QUERY_KEY,
+} from "serviceFacades/filesystem";
+
+import { CircularProgress, InputAdornment } from "@material-ui/core";
+
+export default function DiskResourceSelector({ param, ...props }) {
+    const {
+        field: { value },
+    } = props;
+
+    const [statError, setStatError] = React.useState();
+
+    React.useEffect(() => {
+        if (!value) {
+            setStatError(null);
+        }
+    }, [value]);
+
+    const { t } = useTranslation("launch");
+
+    const { isFetching: isFetchingStat } = useQuery({
+        queryKey: [DATA_DETAILS_QUERY_KEY, { paths: [value] }],
+        queryFn: getResourceDetails,
+        config: {
+            enabled: value,
+            onSuccess: () => {
+                setStatError(null);
+            },
+            onError: (error) => {
+                if (ERROR_CODES.ERR_DOES_NOT_EXIST === getErrorCode(error)) {
+                    setStatError(t("errorResourceDoesNotExist"));
+                } else {
+                    setStatError(null);
+                }
+            },
```

Comment:
This is the current behavior in old DE, where only 404 errors are explicitly checked, and everything else is ignored.

Reply:
🤔 Seems strange that we would ignore other errors, but if it's been that way in the old DE for however long, I'm guessing it's fine or for a reason. Any final blocking errors would pop up after they tried to launch the app anyway, I assume.

Reply:
Yeah, this was odd, but I did some testing in old DE, and it seems there can be a case where I can share a private app with `write` perms with you, and if it uses a default input that you don't have read permissions on, it will still display in the app editor without an error message for you (even though this check will return an `ERR_NOT_READABLE`).

I suppose that still makes sense, since the file does exist, only it won't pre-populate in the app launch form if you try to run it.

I did have a generic error message for other errors (such as "Error loading file/folder details. If this path does not exist, the app may not run successfully"), but I removed it in the last commit since old DE doesn't bother 🤷